File: src/main/java/tomeko/legacyskyblock/hud/ActionBar.java
package tomeko.legacyskyblock.hud;

import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents;
import net.minecraft.network.chat.Component;
import org.polyfrost.oneconfig.api.config.v1.annotations.Switch;
import tomeko.legacyskyblock.config.LegacySkyblockConfig;
import tomeko.legacyskyblock.utils.HypixelPackets;

public class ActionBar {
    @Switch(
            title = "Hide Defense",
            category = LegacySkyblockConfig.CATEGORY_SCREEN,
            subcategory = "Action Bar"
    )
    private boolean hideDefense = false;

    @Switch(
            title = "Hide True Defense",
            category = LegacySkyblockConfig.CATEGORY_SCREEN,
            subcategory = "Action Bar"
    )
    private boolean hideTrueDefense = false;

    @Switch(
            title = "Hide Florid Zombie Sword's Charges",
            category = LegacySkyblockConfig.CATEGORY_SCREEN,
            subcategory = "Action Bar"
    )
    private boolean hideFloridZombieSwordsCharges = false;

    public ActionBar() {
        ClientReceiveMessageEvents.MODIFY_GAME.register((message, fromActionBar) -> {
            if (!fromActionBar || !HypixelPackets.inSkyblock) {
                return message;
            }

            //Hide True Defense
            if (hideTrueDefense) {
                message = replaceActionBar(message, "§f.*?§f❂ True Defense");
            }

            //Hide Defense
            if (hideDefense) {
                message = replaceActionBar(message, "§a.*?§a❈ Defense");
            }

            //Hide Florid Zombie Sword's charges
            if (hideFloridZombieSwordsCharges) {
                message = replaceActionBar(message, "§e§lⓩⓩⓩⓩⓩ§6§l");
                message = replaceActionBar(message, "§e§lⓩⓩⓩⓩ§6§lⓄ");
                message = replaceActionBar(message, "§e§lⓩⓩⓩ§6§lⓄⓄ");
                message = replaceActionBar(message, "§e§lⓩⓩ§6§lⓄⓄⓄ");
                message = replaceActionBar(message, "§e§lⓩ§6§lⓄⓄⓄⓄ");
                message = replaceActionBar(message, "§e§l§6§lⓄⓄⓄⓄⓄ");
            }
            return message;
        });
    }

    private Component replaceActionBar(Component message, String replace) {
        String blank = " {5}";
        message = Component.nullToEmpty(message.getString().replaceAll(replace + blank, ""));
        message = Component.nullToEmpty(message.getString().replaceAll(blank + replace, ""));
        message = Component.nullToEmpty(message.getString().replaceAll(replace, ""));
        return message;
    }
}

File: src/main/kotlin/tomeko/legacyskyblock/hud/ActionBar.kt
package tomeko.legacyskyblock.hud

import net.fabricmc.fabric.api.client.message.v1.ClientReceiveMessageEvents
import net.minecraft.network.chat.Component
import tomeko.legacyskyblock.config.LegacySkyblockConfig
import tomeko.legacyskyblock.utils.HypixelPackets

object ActionBar {
    fun register() {
        ClientReceiveMessageEvents.MODIFY_GAME.register(ClientReceiveMessageEvents.ModifyGame { message: Component, fromActionBar: Boolean ->
            var message = message
            if (!fromActionBar || !HypixelPackets.inSkyblock) {
                return@ModifyGame message
            }

            //Hide True Defense
            if (LegacySkyblockConfig.actionBarHideTrueDefense) {
                message = replaceActionBar(message, "§f\\d+.")
            }

            //Hide Defense
            if (LegacySkyblockConfig.actionBarHideDefense) {
                message = replaceActionBar(message, "§a\\d+.")
            }

            return@ModifyGame message
        })
    }

    private fun replaceActionBar(message: Component, replace: String): Component {
        var message = message
        val blank = " {5}"
        message = Component.nullToEmpty(message.string.replace((replace + blank).toRegex(), ""))
        message = Component.nullToEmpty(message.string.replace((blank + replace).toRegex(), ""))
        message = Component.nullToEmpty(message.string.replace(replace.toRegex(), ""))
        return message
    }
}
